### ISSUE TITLE
Building Error

### ISSUE THREAD

**Issue by vg0426** [mapsme/omim#4743]
Can someone PLEASE update the install instructions file. I really would like to use this project for an android app I want to build but so far I've tried to compile this project in Ubuntu 14.04, Debian 8.6 and Fedora 23 with no success at all. I know nothing of C therefore I can't even understand what could be the source of the problem. All I want is to be able to compile this project so that I can be able to use it in android. Please someone help me

**Comment by stevegdavis** [comment]
You need to state in detail the problem such as a log of the build error you are getting and post it here then you may get a response from the developers.

**Comment by Zverik** [comment]
Agreed, withot an error message we cannot tell what's wrong. Most likely you are missing some of the libraries, or forgot to run `set_up_android.py`.

**Comment by stevegdavis** [comment]
@Zverik I assume that people have built under Ubuntu 14.04, Debian 8.6 and Fedora 23 successfully ?

**Comment by Zverik** [comment]
I built it under Ubuntu 14.04 and Fedora 20-something, yes. Also under CentOS 6 and 7.

**Comment by vg0426** [comment]
Ok here it goes then. I've installed Ubuntu 14.04, Debian 8.6 and Fedora 23 on VMWare VMs and started with the Ubuntu VM and was getting errors in compilation but then I read one the issues posted here on github of someone saying that on ubuntu was giving problem all the time but in debian wasn't, so then i moved to Debian and got the same I got on Fedora (will describe bellow) with the zlib.h file not found, this is the error log recorded by the compilation:

**/home/vgonzalez/Projects/omim/3party/freetype/src/gzip/ftgzip.c:50:10: error: 'zlib.h' file not found with <angled> include; use "quotes" instead
#include <zlib.h>
         ^~~~~~~~
         "zlib.h"
1 error generated.
make[2]: *** [../../out/release/tmp/freetype/ftgzip.o] Error 1
make[1]: *** [sub-freetype-make_first] Error 2
make: *** [sub-3party-make_first] Error 2
make: *** Waiting for unfinished jobs....
In file included from /home/vgonzalez/Projects/omim/coding/zip_creator.cpp:17:
/home/vgonzalez/Projects/omim/3party/minizip/zip.h:50:10: fatal error: 'zlib.h' file not found
#include "zlib.h"
         ^
1 error generated.
make[1]: *** [../out/release/tmp/coding/zip_creator.o] Error 1
make[1]: *** Waiting for unfinished jobs....
In file included from /home/vgonzalez/Projects/omim/coding/zip_reader.cpp:10:
/home/vgonzalez/Projects/omim/3party/minizip/unzip.h:51:10: fatal error: 'zlib.h' file not found
#include "zlib.h"
         ^
1 error generated.
make[1]: *** [../out/release/tmp/coding/zip_reader.o] Error 1
make: *** [sub-coding-make_first] Error 2**

So then I started reading to find any issue reported related to that error and came with this issue https://github.com/mapsme/omim/issues/3808, I followed the instructions posted on the issue and manage to solve the problem but then I got another problem related to opening hours when I ran the compiling command for fedora "SPEC=linux-clang tools/unix/build_omim.sh -cr" but then I ran the compile command once again only this time like root "sudo SPEC=linux-clang tools/unix/build_omim.sh -cr" and no errors where found. After this I started doing the android build instructions, downloaded the android sdk and ndk, ran the "./omim/tools/android/set_up_android.py" script and after that executed the build command with gradle according to the instructions "./omim/android/gradlew clean assembleWebRelease" and this is the new error am getting:

**Starting a new Gradle Daemon for this build (subsequent builds will be faster).
Incremental java compilation is an incubating feature.
:ndkBuildClean
/home/vgonzalez/Android/Sdk/ndk-bundle/build/core/setup-app.mk:120: *** Android NDK: Aborting    .  Stop.
Android NDK: armeabi-v7a-hard is no longer supported. Use armeabi-v7a.    
Android NDK: See https://android.googlesource.com/platform/ndk/+/master/docs/HardFloatAbi.md    
Android NDK: NDK Application 'local' targets unknown ABI(s): armeabi-v7a-hard    
Android NDK: Please fix the APP_ABI definition in ./jni/Application.mk    
:ndkBuildClean FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':ndkBuildClean'.
> Process 'command '/home/vgonzalez/Android/Sdk/ndk-bundle/ndk-build'' finished with non-zero exit value 2

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.

BUILD FAILED

Total time: 1 mins 7.29 secs**

Could you please help me with this error now?
All the tests on the different OS has been with freshly installed versions of them in VMWare VMs, that's is why I was asking in my first post that the install instructions to be updated. Thanks for your cooperation and please help me with the new error am getting.

**Comment by stevegdavis** [comment]
Are running VMWare under Windows ? if so why not try to build it under Windows directly ? unfortunately maps.me do not support windows build but it is possible, I managed to build under windows 7 about a year ago but it is quite a lengthy task, so in your case still problematic !.

**Comment by mnemonicflow** [comment]
@vg0426 In newer Android NDK versions, like android-ndk-r12b, android-ndk-r13b, the support for the ARM hard float ABI has been removed. 
Therefore as the NDK error message says, you have to remove the "-hard" part from the ARM ABI specified in the following two files:
```
build.gradle
330:def archs = ['x86', 'armeabi-v7a-hard']

jni/Application.mk
11:  APP_ABI := armeabi-v7a-hard x86
```

However, you also have to change the paths to the include headers and to clang compiler because they have also changed in the newer NDK's.
